### PR TITLE
Move js logic & disable polling on error

### PR DESCRIPTION
**Pull request by epinxteren**
Test scenario:

- Registreer en wacht totdat sessie verloop.
  - Session expired
  - Polling moet uit staan.
- Ga offline (in browser)
  - Fout melding scherm
  - Polling moet uit staan.

Je kan ook testen met js console met:

registrationStateMachine.statusReceivedHandler('3') // Tot 1-5

See: https://www.pivotaltracker.com/story/show/160949726

### CODE REVIEW

File: src/AppBundle/Resources/views/default/registration.html.twig [hunk 1]
```diff
@@ -33,81 +33,18 @@
...
             {{ 'enrol.status.error' | trans }}
+            <a href="{{  path('app_identity_registration') }}">{{ 'enrol.retry' | trans }}</a>.
         </div>
     </div>
     <div class="content-container qr">
         <img src="{{ url('app_identity_registration_qr') }}">
         {{ 'enrol.download' | trans | raw }}
     </div>
-
+    <script src="{{ asset('build/registration.js') }}" ></script>
```
Comment (MKodde) [inline src/AppBundle/Resources/views/default/registration.html.twig]: :+1: for moving more inline JS to external TS files!

File: src/AppBundle/Resources/javascript/RegistrationStateMachine.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { StatusPollService } from './StatusPollService';
+import { RegistrationStatusComponent } from './Component/RegistrationStatusComponent';
+import { Component } from './Component/Component';
+
+export class RegistrationStateMachine {
+  private IDLE = '1';
```
Comment (MKodde) [inline src/AppBundle/Resources/javascript/RegistrationStateMachine.ts]: Why are these status codes strings?

Comment (epinxteren) [reply src/AppBundle/Resources/javascript/RegistrationStateMachine.ts]: The status codes are from tiqr library, maybe we should have converted the codes inside the Tiqr legacy adapter. Don't think it's worth the effort for now.

**Comment by epinxteren** [comment]
Could not live with the fact that there where no test, created test for the state machine.

**Pull request merged**